src/sage/schemes/elliptic_curves/ell_number_field.py

```diff
@@ -4062,3 +4067,246 @@ def rational_points(self, **kwds):
+
+            sage: E = EllipticCurve('11a1')
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over QQ are modular
+            True
+            sage: K.<a> = QuadraticField(5)
+            sage: E = EllipticCurve([0,1,0,a,0])
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over real quadratic fields are modular
+            True
+            sage: E = EllipticCurve([0,0,0,a,0])
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over real quadratic fields are modular
+            True
+            sage: K.<a> = CyclotomicField(5)
+            sage: E = EllipticCurve([0,1,0,a,0])
+            sage: E.is_modular(verbose=True)
+            Unable to determine modularity except over totally real and imaginary quadratic fields
+            False
+
+        Some examples from the LMFDB.  Over a totally real cubic field::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([-3, -7, -1, 1]))
+            sage: E = EllipticCurve([K([-4,-1,1]),K([-4,-2,1]),K([-4,-1,1]),K([-2807,-660,446]),K([5874,1376,-933])])
+            sage: E.is_modular(verbose=True)
+            Modular since the mod 3 Galois image is neither reducible nor split
+            True
+
+        Over imaginary quadratic fields `K` over which `X_0(15)` has
+        positive rank, there are many curves where current results do
+        not imply modularity::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([2, -1, 1]))
+            sage: EllipticCurve('15a1').change_ring(K).rank()
+            1
+            sage: E = EllipticCurve([K([1,0]),K([1,-1]),K([1,1]),K([-105,98]),K([-615,1188])])
+            sage: E.is_modular(verbose=True)
+            Modularity not established: this curve has small image at 3 and 5
+            False
+            sage: EllipticCurve('15a1').change_ring(K).rank()
+            1
+
+        Nevertheless, over the same field, most curves pass::
+
+            sage: E = EllipticCurve([K([0,0]),K([0,1]),K([1,1]),K([-8,-3]),K([-5,-5])])
+            sage: E.is_modular(verbose=True)
+            Modular since the mod 3 Galois image is neither reducible nor split
+            True
+
+        Over a field which is neither totally real nor imaginary
+        quadratic, no conclusion is currently possible::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([1, 0, -1, 1]))
+            sage: K.is_totally_real()
+            False
+            sage: E = EllipticCurve([K([0,0,1]),K([1,0,0]),K([0,0,1]),K([0,0,0]),K([0,0,0])])
+            sage: E.is_modular(verbose=True)
+            Unable to determine modularity except over totally real and imaginary quadratic fields
+            False
+
+        """
+        def not_from(group, p, j):
+            return len(_modular_polynomial(group, p, j).roots()) == 0
```

Comment:
```suggestion
            return not _modular_polynomial(group, p, j).roots()
```
This should be slightly faster.